File: scripts/test_upload.py
import os
import urllib.request

env = {}
with open('.env.local') as f:
    for line in f:
        line = line.strip()
        if line and not line.startswith('#') and '=' in line:
            k, v = line.split('=', 1)
            env[k.strip()] = v.strip().strip('"').strip("'")

supabase_url = env.get('NEXT_PUBLIC_SUPABASE_URL')
service_key = env.get('SUPABASE_SERVICE_ROLE_KEY')

filepath = 'data/notes_markdown/entrance-biology.md'
with open(filepath, 'rb') as f:
    data = f.read()

url = f"{supabase_url}/storage/v1/object/study_notes/entrance-biology.md"
req = urllib.request.Request(url, data=data, method='POST')
req.add_header('Authorization', f'Bearer {service_key}')
req.add_header('apikey', service_key)
req.add_header('Content-Type', 'text/markdown')
req.add_header('x-upsert', 'true')

try:
    with urllib.request.urlopen(req) as response:
        print("Status:", response.status)
        print("Body:", response.read().decode())
except urllib.error.HTTPError as e:
    print("Error Status:", e.code)
    print("Error Body:", e.read().decode())

File: scripts/test_upload.js
const fs = require('fs');
const path = require('path');
require('dotenv').config({ path: '.env.local' });
const SUPABASE_URL = process.env.NEXT_PUBLIC_SUPABASE_URL;
const SERVICE_ROLE_KEY = process.env.SUPABASE_SERVICE_ROLE_KEY;

async function run() {
  const file = 'entrance-biology.md';
  const filePath = path.join(__dirname, 'data', 'notes_markdown', file);
  const content = fs.readFileSync(filePath);
  
  const uploadRes = await fetch(`${SUPABASE_URL}/storage/v1/object/study_notes/${file}`, {
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${SERVICE_ROLE_KEY}`,
      'Content-Type': 'text/markdown',
      'x-upsert': 'true'
    },
    body: content
  });
  console.log('Upload status:', uploadRes.status);
  console.log('Upload response:', await uploadRes.text());
}
run();
